signature-specification.md

```diff
@@ -0,0 +1,228 @@
+
+To leverage JWS claims validation functionality already provided by libraries, we have defined `iat`, `exp` as top-level nodes.
+
+**ProtectedHeaders**:
+Notary v2 uses only 3 protected headers: alg, cty, and crit.
+```json
+{
+   "alg": "RS256",
+   "cty": "application/vnd.cncf.notary.v2.jws.v1",
+   "crit":["cty"]
+}
+```
+* **`alg`**: JWS needs algorithm(alg) to be present in the header, so we have added it as a protected header.
+* **`cty`**: Content type(cty) used to declare the media type of the secured content(the payload). This will be used to version different variotions of JWS signature. The supported value is `application/vnd.cncf.notary.v2.jws.v0`.
+* **`crit`**: Indicates the list of headers that implementation MUST be understood and processed. The value MUST be `["cty"]`.
+
+**UnprotectedHeaders**:
+Notary v2 uses only 2 unprotected headers: timestamp and x5c.
+```
+{
+   "timestamp": "<Base64Url(TimeStampToken)>",
+   "x5c": ["<Base64(DER(leafCert))>", "<Base64(DER(intermediateCACert))>", "<Base64(DER(rootCert))>"]
+}
+```
+* **`timestamp`**: Used to store time stamp token. Only [RFC3161]([rfc3161](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2)) compliant TimeStampToken are supported. Use of this header is OPTIONAL.
+* **`x5c`**: Contains the X.509 public key certificate or certificate chain([RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)) corresponding to the key used to digitally sign the JWS.
+
+**Signature**:
+In JWS signature is calculated by combining JWSPayload and protected headers. The process is described below:
+1. Compute the base64url value of ProtectedHeaders.
+2. Compute the base64url value of JWSPayload.
+3. Build message to be signed by concatenating the values generated in step 1 and step 2 using '.'
+`ASCII(BASE64URL(UTF8(ProtectedHeaders)) ‘.’ BASE64URL(JWSPayload))`
+4. Compute the signature on the message constructed in the previous step by using the signature algorithm defined in the corresponding header element: alg.
+5. Compute the base64url value of the signature produced in the previous step. This is the value of the signature property used in the signature envelope.
+
+**Signature Envelope**:
+The final signature envelope comprises of Claims, ProtectedHeaders, UnprotectedHeaders, and signature.
+```json
+{
+   "payload": "<Base64Url(JWSPayload)>",
+   "signatures": [
+       {
+           "protected": "<Base64Url(ProtectedHeaders)>",
+           "header": {
+               "timestamp": "<Base64Url(TimeStampToken)>",
+               "x5c": ["<Base64(DER(leafCert))>", "<Base64(DER(intermediateCACert))>", "<Base64(DER(rootCert))>"]
+           },
+           "signature": "Base64Url( sign( ASCII( <Base64Url(JWSPayload)>.<Base64Url(ProtectedHeader)> )))"  
+       }
+   ]
+}
+```
+If a JWS contains only one signature as above, the JWS can be flattened as
```

Comment:
Are there any scenarios where multiple signatures would be pushed together, rather than separate artifacts?

Reply:
As of now we don't have scenario for multiple signature. removed JWS JSON format that supports multiple signature i.e. we only support flattened JWS JSON format.